### DISCUSSION TITLE
WinGet DSC Resource

### DISCUSSION THREAD

**Discussion by denelon** [microsoft/winget-cli#3019] [PowerShell]
Naming things is hard.

We're working on the WinGet DSC resource:
* https://github.com/microsoft/winget-cli/issues/2697

We already use "WinGetPackage" to represent the packages managed by WinGet.

In order to disambiguate from those packages, the suggestion was made to consider "WinGetSoftware" to represent WinGet itself.

The cmdlet names we're considering are:
Assert-WinGetIntegrity (boolean return), but maybe "Get-WinGetSoftware" (tells you if it's installed, and additional data if it is).

We would also likely have "Install-WinGetSoftware" for installing WinGet without the dependency of the Microsoft Store.

**Comment by HotCakeX** [comment]
`Get-WinGetSoftware` is much better and more similar to other PowerShell commands than `Assert-WinGetIntegrity`

**Comment by Trenly** [comment]
`Get` makes me think that it will fetch something, which makes sense if the additional details of `winget info` are provided if installed.

`Assert` makes me think that it will do some action to ensure that WinGetSoftware is installed, and fail if not. I don't think that `Assert` should be used.

Other options could be -
* `Confirm-WinGetSoftware`, but that doesn't seem quite right.
* `Find-WinGetSoftware`, which could work, but `Get` seems better


In terms of installing WinGet, I could maybe also see `Initialize-WinGetSoftware`, as an indication that it may install more than just WinGet itself.

**Comment by tristanbarcelon** [comment]
Is WinGetPackageManager too long because it completely describes what winget-cli is, a package manager.

**Reply by joelst** [reply]
This is a difficult one. I read this two weeks ago and am still trying to solidify my opinions.

After thinking about it for a while, I think that `WinGet` should be fine. It aligns with the DSC resources created in #2697  

However, if you all REALLY feel that you need another noun after WinGet, I think that `WinGetSettings` would also work in most situations. 

I worry that `WinGetSoftware` could be confused with `WinGetPackages.`

If you run `Find-DSCResource` you'll see examples all over the place. You can also find some really good examples at [DSC Community on GitHub](https://github.com/dsccommunity)

###  Here are a couple potentially relevant examples to compare:

- SharePointDSC uses `Export-SPConfiguration` to create a configuration file.
- SecurityPolicyDSC and ConfigMgrCBDsc use `Test-` to validate settings similar to `Test-DscConfiguration` (For some reason I prefer `Assert-`, but I think the majority of cases I found use `Test-` -- please prove me wrong!

**Comment by johlju** [comment]
> Assert-WinGetIntegrity (boolean return)

If a command should return true or false it might be better to name the command `Test-WinGetIntegrity`

**Comment by nickgw** [comment]
Is just `WinGet` too vague? To me it makes sense that `WinGet` configures winget itself, and `wingetpackage` configures packages from the winget provider
